components/esp_netif/lwip/esp_netif_lwip_slip.c

```diff
@@ -0,0 +1,335 @@
+
+        return NULL;
+    }
+#endif
+
+    ESP_LOGI(TAG, "%s: Created SLIP interface (netif %p, slip_ctx: %p)", __func__, esp_netif, slip_ctx);
+
+    return slip_ctx;
+}
+
+/**
+ * @brief Creates new SLIP related structure
+ */
+esp_err_t esp_netif_start_slip(lwip_slip_ctx_t *slip_ctx) {
+    ESP_LOGI(TAG, "%s: Starting SLIP connection (slip_ctx: %p, addr: %s)", __func__, slip_ctx, ip6addr_ntoa(&slip_ctx->addr));
+
+    struct netif * netif_impl = slip_ctx->esp_netif->lwip_netif;
+
+    // Store serial port number in net interface for SIO open command
+    netif_impl->state = (void*)slip_ctx->uart_dev;
+
+    // Allocate RX buffer if not existing
+    if (slip_ctx->buffer == NULL) {
+        slip_ctx->buffer = malloc(slip_ctx->buffer_len);
+    }
+
+#if 1
+    // Attach interface
+    netif_add_noaddr(slip_ctx->esp_netif->lwip_netif, (void*)slip_ctx->uart_dev, &slipif_init, &ip_input);
+
+    // Bind address
+    int8_t addr_index = 0;
+    //netif_add_ip6_address(slip_ctx->esp_netif->lwip_netif, &slip_ctx->addr, &addr_index);
+
+    netif_ip6_addr_set(slip_ctx->esp_netif->lwip_netif, addr_index, &slip_ctx->addr);
+    netif_ip6_addr_set_state(slip_ctx->esp_netif->lwip_netif, addr_index, IP6_ADDR_VALID);
+
+    // Setup interface
+    netif_set_link_up(slip_ctx->esp_netif->lwip_netif);
+    netif_set_up(slip_ctx->esp_netif->lwip_netif);
+#endif
+
+    uart_config_t uart_config = {
+        .baud_rate = slip_ctx->uart_baud,
+        .data_bits = UART_DATA_8_BITS,
+        .parity = UART_PARITY_DISABLE,
+        .stop_bits = UART_STOP_BITS_1,
+        .flow_ctrl = UART_HW_FLOWCTRL_DISABLE,
+    };
+
+    // Initialise uart
+    ESP_ERROR_CHECK(uart_param_config(slip_ctx->uart_dev, &uart_config));
+
+    // Set UART pins
+    ESP_ERROR_CHECK(uart_set_pin(slip_ctx->uart_dev, slip_ctx->uart_tx_pin, slip_ctx->uart_rx_pin, 0, 0));
+
+    // Install UART driver
+    ESP_ERROR_CHECK(uart_driver_install(slip_ctx->uart_dev, slip_ctx->buffer_len, slip_ctx->buffer_len, 10, &slip_ctx->uart_queue, 0));
+
+    // Start slip RX task
+    slip_ctx->running = true;
+    xTaskCreate(esp_slip_uart_rx_task, "slip_rx_task", SLIP_RX_TASK_STACK_SIZE, slip_ctx, SLIP_RX_TASK_PRIORITY, &slip_ctx->rx_task);
+
+    return ESP_OK;
+}
+
+/**
+ * @brief Stops the SLIP interface
+ */
+esp_err_t esp_netif_stop_slip(lwip_slip_ctx_t *slip_ctx) {
+    ESP_LOGI(TAG, "%s: Stopped SLIP connection: %p", __func__, slip_ctx);
+
+    // TODO: stop the interface running
+
+    slip_ctx->running = false;
+
+    vTaskDelete(&slip_ctx->rx_task);
+
+    return ESP_OK;
+}
+
+
+/**
+ * @brief Sets paramaters for the supplied netif
+ */
+esp_err_t esp_netif_slip_set_params(esp_netif_t *netif, const esp_netif_slip_config_t *slip_config) {
+
+    lwip_slip_ctx_t* slip_ctx = netif->lwip_slip_ctx;
+
+    ESP_LOGD(TAG, "%s (slip_ctx: %p)", __func__, slip_ctx);
+
+    if (slip_ctx->running) {
+        ESP_LOGE(TAG, "Cannot set parameters while SLIP interface is running");
+        return -1;
```

Comment:
```suggestion
        return ESP_ERR_INVALID_STATE;
```

or if -1 makes more sense here, `ESP_FAIL` can be used.